src/Middleware/WebSockets/src/WebSocketMiddleware.cs

```diff
@@ -159,7 +163,7 @@ public async Task<WebSocket> AcceptAsync(WebSocketAcceptContext acceptContext)
             }
 
             var key = _context.Request.Headers.SecWebSocketKey.ToString();
```

Comment:
We can avoid the `.ToString()` here for H2

Reply:
The ToString should be harmless in the common case. There should only be one value, and it will return that value without allocating.